### PR TITLE
docs: updated Halshow docs

### PR DESCRIPTION
**Pull request by hansu**
Updated and re-structured the docs about Halshow.
Would be nice if someone could have a quick look at it: https://hansu.github.io/linuxcnc-doc/da71cbdcccfedb06f7ad8ca2beb4e1da6e8c5d63/html/hal/halshow.html

### CODE REVIEW

**Comment by smoe** [comment]
I think this can go in. Maybe the one or other bit I would have phrased differently, but any such editing can be performed later.

I would have made the images smaller. But again - nothing that would need to be addressed now.

**Comment by hansu** [comment]
> I think this can go in. Maybe the one or other bit I would have phrased differently, but any such editing can be performed later.

Let's do it know, that's why I am asking.

> I would have made the images [edit: I mean, the windows that are shown with loads of gray] smaller. But again - nothing that would need to be addressed now.

You mean resizing the images and then make a screenshot, not just scaling them?
I have chosen this size intentionally as it is the default size. And only in the first image there is a lot of gray/white space. The other ones are well filled I think.

File: docs/src/hal/halshow.adoc [hunk 1]
```diff
@@ -53,46 +56,62 @@ $ halshow  --fformat "%.5f" ./my.halshow
...
 
-Clicking on the node name, the word "Components" for example, will
+[[fig:halshow-show-tab]]
+.Halshow: Show Tab
+image::images/halshow-show-tab.png["Halshow: Show Tab",align="center"]
+
+Clicking on the node name, "Components" in the tree for example, will
 show you (under the "Show" tab) all that HAL knows about the contents
-of that node. Figure <<cap:halshow-layout>> shows a list exactly like
+of that node. Figure <<fig:halshow-show-tab, Halshow Show Tab>> shows a list exactly like
```
Comment (smoe) [inline docs/src/hal/halshow.adoc:108]: Please remove the blank behind the comma within the fig ref.

**Review by smoe — APPROVED** [review]
Seems like some other ideas for changes to the text were independent from your changes :) For subsequent editing with git but in particular for the automated translations, please consider to reformat the text to have line breaks in sync with semantics.

**Comment by smoe** [comment]
> You mean resizing the images and then make a screenshot, not just scaling them?

Yes, but you are right, it is only one of the images for which this seems appropriate.

File: docs/src/hal/halshow.adoc [hunk 2]
```diff
@@ -13,18 +13,20 @@
 The script halshow can help you find your way around a running HAL.
 This is a very specialized system and it must connect to a working HAL.
 It cannot run standalone because it relies on the ability of HAL to
-report what it knows of itself through the halcmd interface library. It
-is discovery based. Each time halshow runs with a different LinuxCNC
+report what it knows of itself through the halcmd interface library.
```
[the remark this answers is not in the document]
Comment (hansu) [reply docs/src/hal/halshow.adoc:16]: Where should that be inserted? in this paragraph it doesn't really fit.

[the remark this answers is not in the document]
Comment (smoe) [reply docs/src/hal/halshow.adoc:16]: Rereading the paragraph, I agree. It is am important feature, though, and should be stated somewhere. You'll find a sweet spot, I am sure :)

**Comment by smoe** [comment]
I have looked through you changes - perfectly done! Many thanks!

File: docs/src/hal/halshow.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (hansu) [reply docs/src/hal/halshow.adoc:17]: Ok inserted it in the intro paragraph.

File: docs/src/hal/halshow.adoc [hunk 3]
```diff
@@ -110,27 +130,116 @@ Owner  Type  Dir  Value  Name
...
+
+[[fig:halshow-watch-tab-1]]
+.Halshow: Watch Tab(((Halshow: Watch Tab)))
+image::images/halshow-watch-tab-1.png["Halshow Watch Tab",align="center"]
+
+_Watch_ displays bit type (binary) values using colored circles
+representing LEDs. They show as dark red when a bit signal or pin is
+false, and as light yellow whenever that signal is true. If you select
+a pin or signal that is not a bit type (binary) signal, _watch_ will show
+it as a numerical value. Pins are displayed in black, signals in blue and parameters in brown.
```
Comment (hansu) [inline docs/src/hal/halshow.adoc:164]: @smoe
>please consider to reformat the text to have line breaks in sync with semantics.


You mean like this?
```suggestion
_Watch_ displays bit type (binary) values using colored circles
representing LEDs. 
They show as dark red when a bit signal or pin is false, and as light yellow whenever 
that signal is true.
If you select a pin or signal that is not a bit type (binary) signal, _watch_ will show
it as a numerical value. Pins are displayed in black, signals in blue and parameters in brown.
```

**Pull request merged**